src/include/txn_inline.h

```diff
@@ -501,6 +501,12 @@ __wt_txn_pinned_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *pinned_tsp)
     btree = S2BT_SAFE(session);
     txn_global = &S2C(session)->txn_global;
 
+    /*
+     * There is no need to go further if no pinned timestamp has been set yet.
+     */
+    if (!txn_global->has_pinned_timestamp)
```

Comment:
We should set the pinned timestamp to WT_TS_NONE if txn_global->has_pinned_timestamp is false. We could possibly default the timestamp to WT_TS_NONE before any checks, the side effect of this function would be to always have pinned_ts populated, and the caller need not initialize the variable themselves (or forget to use uninitialized). This function also doesn't return a failure, so it should always populate "*pinned_tsp".

Reply:
Side effects should go before other variable initializations, so :

```
*pinned_tsp = WT_TS_NONE;

btree = S2BT_SAFE(session);
txn_global = &S2C(session)->txn_global;
```